### PR TITLE
Add PPP and update UG

### LINKED ISSUE #206: [PE-D][Tester D] Certain Sentences in the UG have typos: missing periods.
Certain sentences in the User Guide do not end with periods.

For example, under the Feature section:

the first two bullet points have sentences ending with periods, but the subsequent bullet points do not end with periods.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: jamessinmaojun/ped#18

### PR DESCRIPTION
**Pull request by larainezo**
Fixes #206, #200, #144, #213, #182, #146, #143, #212, #209, #176, #172, #170, #150, #174, #171

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -31,273 +49,184 @@ Action     | Format, Examples
...
+Phone Number                         | Must contain 8 digits with no spacing. <br> e.g. `91234567`.
+Email                                | Must contain @ and .com. <br> eg:`dormie@example.com`.
+Room Number                          | {block}-{floor}-{room number}, where block and room number are at least 2 alphanumeric characters and floor is strictly 2 alphanumeric characters. <br> eg:`nw-12-12`.
+Telegram Handle                      | Can only contain case-insensitive letters A-Z, digits 0-9, and underscores, with a length between 5 and 32 characters, and it **should not** be blank. <br> Note: Do not include the `@` symbol in the input. <br> e.g `dormie_123`.
+Birthday                             | `dd/MM/yyyy`, `dd-MM-yyyy`, `yyyy-MM-dd`, `yyyy/MM/dd`. <br> `dd` is the date of the month, `MM` is the month, `YYYY` is the year. <br> e.g. **30/01/2024**, **30-01-2024**, **2024-01-30**, **2024/01/30** all represent 30 January 2024.
+Dorm Tag                             | Any non-empty String. <br> eg. `PGPR`.
+Free Time Tag                        | `DDD:HHmm-HHmm`. <br> `DDD` is from Mon-Sun, `HHmm` is 24 hour time format. <br> e.g. **Mon:1300-1400**.
```
Comment (PallonCX) [inline docs/UserGuide.md:62]: **[Remark for reviewers]**

#206

File: docs/UserGuide.md [hunk 2]
```diff
@@ -31,273 +49,184 @@ Action     | Format, Examples
...
 - `DAY` is from Mon-Sun
 - `TIME` is 24-hour time format
 
-Command usage examples:
-
+Example:
 - `whoisfree Mon:1300`
 
+**Important Note**
+- No 2 people can share the same `phone number`.
```
[the remark this answers is not in the document]
Comment (PallonCX) [reply docs/UserGuide.md:210]: Actually we may also want to specify this for email and telegram

File: docs/UserGuide.md [hunk 3]
```diff
@@ -11,16 +11,34 @@ Dormie is a **desktop app for managing contacts, optimized for use via a Command
...
+...                                 | Ellipsis mean that the field can be used 0, 1, or more times in a single command. <br> e.g `[ft/FREETIMETAG]...` means that the following commands are allowed `ft/Mon:1300-1400`, `ft/Mon:1300-1400 ft/Tue:1400-1600`.
+Order of Parameters                 | Parameters can be in any order. <br> e.g. if the command specifies `n/NAME p/PHONE_NUMBER`, `p/PHONE_NUMBER n/NAME` is also acceptable.
+Extraneous Parameters               | Extraneous parameters for commands that do not take in parameters (such as `list`) will be ignored. <br> e.g. if the command specifies `list 123`, it will be interpreted as `list`.
+Copying and Pasting from this User Guide | If you are using a PDF version of this document, be careful when copying and pasting commands that span multiple lines as space characters surrounding line-breaks may be omitted when copied over to the application.
+
+### Searching for Keywords (Ctrl-F)
+1. Press the Ctrl + F keys on your keyboard.
+2. A search bar or dialog box should appear on your screen. A screenshot is shown below.
+   ![Screenshot of search bar](images/ctrl-F.png)
```
Comment (PallonCX) [inline docs/UserGuide.md:29]: **[Remark for reviewers]**

#176

File: docs/UserGuide.md [hunk 4]
```diff
@@ -11,16 +11,34 @@ Dormie is a **desktop app for managing contacts, optimized for use via a Command
...
+## Command summary
 
 Action     | Format, Examples
 -----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------
-**Add**    | `add n/NAME r/ROOM_NUM​` <br> e.g., `add Alice Lim r/02-03`
+**Add**    | `add n/NAME p/PHONE_NUMBER [e/EMAIL] [r/ROOM_NUMBER] [t/TELEGRAM_HANDLE] [b/BIRTHDAY] [ft/FREE_TIME]...` <br> e.g., `add n/Alice Lim p/91234567`
 **Clear**  | `clear`
 **Delete** | `delete INDEX`<br> e.g., `delete 3`
-**Edit**   | `edit INDEX [n/NAME] [r/ROOM_NUM​]​`<br> e.g.,`edit 1 n/Alex r/05-11`
+**Edit**   | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [r/ROOM_NUM] [t/TELEGRAM] [d/DORM_TAG] [ft/FREE_TIME_TAG]`<br> e.g.,`edit 1 n/Alex p/98765432`
```
Comment (PallonCX) [inline docs/UserGuide.md:41]: **[Remark for reviewers]**

#150  
#174

File: docs/UserGuide.md [hunk 5]
```diff
@@ -11,16 +11,34 @@ Dormie is a **desktop app for managing contacts, optimized for use via a Command
...
+
 ---
 
-## Command Summary
+## Command summary
 
 Action     | Format, Examples
 -----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------
-**Add**    | `add n/NAME r/ROOM_NUM​` <br> e.g., `add Alice Lim r/02-03`
+**Add**    | `add n/NAME p/PHONE_NUMBER [e/EMAIL] [r/ROOM_NUMBER] [t/TELEGRAM_HANDLE] [b/BIRTHDAY] [ft/FREE_TIME]...` <br> e.g., `add n/Alice Lim p/91234567`
```
Comment (PallonCX) [inline docs/UserGuide.md:38]: **[Remark for reviewers]**

#171

**Review by PallonCX — COMMENTED** [review]
Most of the issues are tackled well. I'm mainly concerned about the purpose of reordering some sections, and the change of structure or terms in the section, that cause some information to be lost or become ambiguous.

File: docs/UserGuide.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (larainezo) [reply docs/UserGuide.md:41]: Changed!

File: docs/UserGuide.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (larainezo) [reply docs/UserGuide.md:38]: I'll change it to lexicographical order

File: docs/UserGuide.md [hunk 6]
```diff
@@ -31,273 +49,184 @@ Action     | Format, Examples
...
-
-Format: `addTime INDEX ft/FREE_TIME_TAG...`
-
-Notes on format:
-
-- The index refers to the index number shown in the displayed person list
-- The index must be a positive integer e.g. 1, 2, 3
-
-Command usage examples:
+Creates a new contact for a dorm mate.
```
[the remark this answers is not in the document]
Comment (larainezo) [reply docs/UserGuide.md:114]: Changed

File: docs/UserGuide.md [hunk 7]
```diff
@@ -31,273 +49,184 @@ Action     | Format, Examples
...
-Future enhancement (not implemented yet):
-
-- The command can be performed on multiple person,
-  e.g. `deleteTime 1 2 ft/Mon:1300-1400` will delete the free time tag from both the first and second person.
-
-### Editing a person : edit
-
-Replaces the specified field(s) of an existing person in Dormie with the new input(s).
+- Edit function will replace the specified field with the new input
+- Edits the person at the specified INDEX. The index refers to the index number shown in the displayed person list. The index must be a positive integer 1, 2, 3, …​
```
[the remark this answers is not in the document]
Comment (larainezo) [reply docs/UserGuide.md:148]: changed

File: docs/UserGuide.md [hunk 8]
```diff
@@ -31,273 +49,184 @@ Action     | Format, Examples
...
+Show a link to the User Guide.
 
-View all persons with name containing the given keyword. 
+### Listing all persons : `list`
 
-Format: `find KEYWORD`
+List all contacts and their details
 
-### Provide more detail on existing commands : help
+### Adding Free Time Tag : `addTime`
```
[the remark this answers is not in the document]
Comment (larainezo) [reply docs/UserGuide.md:175]: changed

File: docs/UserGuide.md [hunk 9]
```diff
@@ -31,273 +49,184 @@ Action     | Format, Examples
...
-  e.g. if the command specifies `n/NAME p/PHONE_NUMBER`, `p/PHONE_NUMBER n/NAME` is also acceptable.
-
-- Extraneous parameters for commands that do not take in parameters (such as `list`) will be ignored.<br>
-  e.g. if the command specifies `list 123`, it will be interpreted as `list`.
-
-- Fields with the format followed by `...` can have more than one copy in the command. <br>
-  e.g. `[ft/FREE TIME TAG]...` means there can be more than one free time tag. 
-
-- Command keywords for the commands should be exactly the same as demonstration, or else Dormie would not recognize it.
-  e.g. `add` is the command keyword in the add command, where `Add` or `ADD` will not work.
```
[the remark this answers is not in the document]
Comment (larainezo) [reply docs/UserGuide.md:138]: Added

File: docs/UserGuide.md [hunk 10]
```diff
@@ -31,273 +49,184 @@ Action     | Format, Examples
...
+Example:
 - `whoisfree Mon:1300`
 
+**Important Note**
+- No 2 people can share the same `phone number`.
+- If `freeTimeTags` are edited, the person's `freeTimeTags` will be replaced with the new set of `freeTimeTags`.
+- Example:
+    - Let Joe have a `freeTimeTag`:`Mon:1300-1400` and have the index 1:
+    - `edit 1 ft/`: Will delete the existing `freeTimeTags`
+    - `edit 1 ft/Tue:1300-1400 ft/Wed:1300-1400` Will replace the existing _Monday_ tag with the _Tuesday_ and _Wednesday_ tag.
```
[the remark this answers is not in the document]
Comment (larainezo) [reply docs/UserGuide.md:215]: This is part of the features section

File: docs/UserGuide.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (larainezo) [reply docs/UserGuide.md:210]: Specified

**Review by donwong2308 — APPROVED** [review]
LGTM

**Review by nyankoclaws — APPROVED** [review]
LGTM

**Review by PallonCX — APPROVED** [review]
LGTM

**Review by JustWeiHao — APPROVED** [review]
LGTM

**Comment by donwong2308** [comment]
Closed #197 #198 #149

**Pull request merged**